### PR TITLE
First part of fire line functionality - Rivers

### PR DESCRIPTION
**Pull request by dstrawberrygirl**
This PR deviated somewhat from the initial story, and as such now includes:

- The remaining visual pieces for drought conditions, terrain color now reflects drought level
- Adjustments to moisture levels as per the table in https://www.pivotaltracker.com/story/show/165452066
- Refactoring for drought vs moisture
- Rivers do not burn, and have some effect in slowing down the spread of fire, though additional work is needed to accommodate wind strength / direction and preventing the spread of fire
- Simulation edges no longer burn

### CODE REVIEW

File: src/models/fire-model.ts [hunk 1]
```diff
@@ -75,6 +82,14 @@ const FuelConstants: {[key in LandType]: Fuel} = {
   }
 };
 
+// values for each level of vegetation: Grass, Shrub, ForestSmallLitter, ForestLargeLitter
+export const moistureLookups: {[key in DroughtLevel]: number[]} = {
+  [DroughtLevel.NoDrought]: [0.1275, 0.255, 0.17, 0.2125],
+  [DroughtLevel.MildDrought]: [0.09, 0.18, 0.12, 0.15],
+  [DroughtLevel.MediumDrought]: [0.0525, 0.105, 0.07, 0.0875],
+  [DroughtLevel.SevereDrought]: [0.015, 0.03, 0.02, 0.025],
```
Comment (pjanik) [inline src/models/fire-model.ts]: Out of curiosity, where are these values coming from? Based on experiments or was that provided by someone from the project team? 

I guess Rothermal model specifies only moisture content for given vegetation type and there's no concept of drought level, right?

File: src/components/drought-selector.tsx [hunk 2]
```diff
@@ -1,28 +1,29 @@
...
 import { Slider } from "@material-ui/core";
 import VerticalHandle from "../assets/slider-vertical.svg";
-import NoDrought from "../assets/terrain/drought-no.svg";
-import MildDrought from "../assets/terrain/drought-mild.svg";
-import MedDrought from "../assets/terrain/drought-med.svg";
-import SevereDrought from "../assets/terrain/drought-severe.svg";
 
 import { generateMarks, droughtIcons, droughtLabels } from "./vertical-selectors";
 import * as css from "./vertical-selectors.scss";
+import { DroughtLevel } from "../models/fire-model";
```
Comment (pjanik) [inline src/components/drought-selector.tsx]: In general, we need `DroughtLevel` as the same levels will result in different moisture content for different vegetation types, right?

File: src/models/fire-model.ts (hunk 1 above)
Comment (pjanik) [reply src/models/fire-model.ts]: Ah, I see the table in PT now.

**Comment by dstrawberrygirl** [comment]
Thanks for the review, @pjanik - yes you're correct this did deviate from the original intent... tackling fire lines proved harder than anticipated due to the "many cells ahead" approach. I had some thoughts on how to approach it - perhaps it's simple enough to take the cells that are the fire line / river, then somehow figuring out what side the spark is on (tricky, especially since we can have multiple sparks), and working on adjusting the cells on the "protected" side to reduce burn potential unless a spark breaches the line due to wind. I haven't done as much with the model itself, so hopefully we can catch up next week and figure out what might work.

**Pull request merged**